### PR TITLE
Refactor validate and verify

### PR DESCRIPTION
**Pull request by MattiasGees**
**What this PR does / why we need it**:
Make sure that validate and verify are present on cluster, environment and provider. This streamlines the use of both validate and verify as central calling point instead of all different methods.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)
I need this for #276 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/tarmak/provider/amazon/amazon.go [hunk 1]
```diff
@@ -356,6 +356,10 @@ func (a *Amazon) Validate() error {
 
 }
 
+func (a *Amazon) Verify() (result error) {
+	return result
```
Comment (charlieegan3) [inline pkg/tarmak/provider/amazon/amazon.go]: Am I right in saying this is the same as returning nil?

Comment (charlieegan3) [reply pkg/tarmak/provider/amazon/amazon.go]: I am https://play.golang.org/p/vJ5A24UnA99

I'm going to stick my neck out and say I prefer `return nil` :fire: 

```
func Verify() (error) {
	return nil
}
```

Comment (charlieegan3) [reply pkg/tarmak/provider/amazon/amazon.go]: (this all looks fine. I'm winding down for the day. Just ignore my wittering...)

Comment (MattiasGees) [reply pkg/tarmak/provider/amazon/amazon.go]: Yeah we use the same methodology on other places as well.

**Pull request merged**